### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by principlebrothers**
## Description
In this project my partner @alanluqman  and I @principlebrothers created `unit_testing.rb` and wrote unit test for all the library app classes mentioned below.

### School library app classes
- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`
- all classes that was created while implementing the Decorator design pattern

### CODE REVIEW

File: unit_testing.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  it 'Test Student attributes' do
+    student = Student.new(17, 4, 'Jan', true)
+
+    expect(student.name).to eq 'Jan'
+    expect(student.age).to eq 17
+    expect(student.classroom).to eq 4
+    expect(student.can_use_servises?).to be_truthy
+    expect(student).to be_an_instance_of(Student)
+  end
+end
```
Comment (alexander16108) [inline unit_testing.rb:41]: Optional - 
I suggest you create more test cases to practice unit testing in rails to better prepare you and strengthen you knowledge on unit testing 😄 👍🏾 💯

**Review by alexander16108 — APPROVED** [review]
Hi Team 👋🏾 [@principlebrothers  and @alanluqman ], 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- passing linters  ✔️ 
- passing tests ✔️ 
- clean code ✔️ 
- good pr✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: unit_testing.rb (hunk 1 above)
Comment (alanluqman) [reply unit_testing.rb:41]: Thanks for your review Dear @alexander16108

Comment (principlebrothers) [reply unit_testing.rb:41]: Thanks for the suggestion @alexander16108

**Pull request merged**